unicodetools/src/main/java/org/unicode/idna/Uts46.java

```diff
@@ -199,6 +199,9 @@ public static boolean hasBidiError(String label, Set<Errors> errors) {
         final boolean LTR = L.contains(firstChar);
         if (!RTL && !LTR) {
             errors.add(Errors.B1);
+            
+            // A label that fails B1 can't fail any of B2 to B6.
+            return true;
```

Comment:
```suggestion
            // A label that fails B1 can't fail any of B2 to B6.
            return errors.size() > oldErrorLength;
```
The premise makes sense to me, but I think the return should be as if we reached the end of the function without this shortcut.

Reply:
Done.